src/components/TokenRow/TokenRow.tsx

```diff
@@ -55,6 +59,8 @@ export function TokenRow<C>({
       )}
       <Controller
         render={({ field: { onChange, value } }) => {
+          if (typeof value !== 'object' && value !== null) throw new TypeError('value')
+          assume<string | number | undefined | null>(value)
```

Comment:
So this could be done like `const castValue = value as string | number | undefined | null` as well, but IMO the `assume()` function is cleaner -- except that we need somewhere better to put it, like a utils library or something.

The underlying issue is something in `react-hook-form`'s `UnpackNestedValue` type; the closest thing I can find is https://github.com/react-hook-form/react-hook-form/issues/2978, but it doesn't look like the fix there fixes this.

Reply:
Sorted, with some inspiration from https://github.com/react-hook-form/react-hook-form/pull/6562.